src/app/loginComponents/accounts/internal/change-username/change-username.component.html

```diff
@@ -1,26 +1,52 @@
-<p *ngIf="showText">
-  Your current username is <strong>{{ user?.username }}</strong
-  >. It may have been auto-generated when you initially created a Dockstore account. You can update your username only if you meet certain
-  conditions.
-</p>
+<h1 mat-dialog-title *ngIf="!noDialog">Edit Dockstore Username</h1>
 
 <p>
-  You <strong>can<span *ngIf="(canChangeUsername$ | async) === false">'t</span></strong> change your username.
+  Your current username is <strong>{{ user?.username }}</strong
+  >. It may have been auto-generated when you initially created a Dockstore account.
+  <span *ngIf="(canChangeUsername$ | async) === true; else cannotChange"
+    >You <strong>can</strong> update your username since you have no published workflows/tools and have shared no workflows via
+    permissions.</span
+  >
+  <ng-template #cannotChange
+    >You <strong>cannot</strong> update your username since you have published workflows/tools or have shared workflows via
+    permissions.</ng-template
+  >
 </p>
 
-<mat-card class="alert alert-warning" role="alert" *ngIf="showEmailWarning">
-  <mat-icon>warning</mat-icon> Your username seems to be an email. We recommend changing your username to something else so that your email
-  is not picked up by a spambot. The @ symbol is not allowed in a username, so note that once you change your username you cannot change it
-  back to your email.
+<!-- Alert -->
+<mat-card *ngIf="showEmailWarning" class="alert alert-danger mat-elevation-z p-2" role="alert" fxLayout fxLayoutGap="1rem">
+  <mat-icon class="alert-danger-icon">warning</mat-icon>
+  <div>
+    <strong>Caution:</strong> Your username seems to be an email address. We recommend changing your username to something else so that your
+    email is not picked up by a spambot. The @ symbol is not allowed in a username, so note that once you change your username you cannot
+    change it back to your email.
+  </div>
 </mat-card>
 
-<mat-card class="alert alert-warning" role="alert" *ngIf="usernameChangeRequired">
-  <mat-icon>warning</mat-icon>Your username contains one or more banned words. Some operations on Dockstore will be blocked until you change
-  your username.
+<!-- Warnings -->
+<mat-card
+  *ngIf="(canChangeUsername$ | async) === true && !noDialog"
+  class="alert alert-warning mat-elevation-z p-2"
+  role="alert"
+  fxLayout
+  fxLayoutGap="1rem"
+>
+  <mat-icon class="alert-warning-icon">warning</mat-icon>
+  <div>
+    <strong>Warning:</strong> You may have unpublished workflows or tools. Changing your username could impact any shared unpublished
```

Comment:
I'm a little confused by this message. You can't share unpublished workflows/tools, although there is the edge case of sharing via permissions.  What case is it covering? (I may see the answer later in the code).

Reply:
I think it really was that edge case of sharing via permissions. Maybe we were worried that if you change your name, SAM-based sharing would break.

Reply:
> I think it really was that edge case of sharing via permissions. Maybe we were worried that if you change your name, SAM-based sharing would break.

Ah, right. If a hosted workflow's "namespace" (username) changes, it would break SAM-based sharing.